802.1x: Hide the password from output by default

cathay4t opened the pull request:
Introduced new argument `include_secret=False|True` to:
    * `libnmstate.show_running_config()`
    * `libnmstate.show()`

which will mask the passwords with `<_password_hid_by_nmstate>` by
default(`include_secret=False`).

The passwords are also masked in logging and error message.

Added new argument:
    * `nmstatectl show -s `

Without `-s` switch, nmstatectl output will use the masked password.

Integration test case updated.
Unit test cases included.

cathay4t commented:
The work is done by the top level so the plugin does not have to worry about it.

ffmancera commented on libnmstate/netapplier.py:
```diff
@@ -65,7 +66,10 @@ def apply(
     """
     logging.debug(f"Nmstate version: {get_version()}")
-    logging.debug(f"Applying desire state: {desired_state}")
+
+    desired_state_without_secrets = copy.deepcopy(desired_state)
+    hide_the_secrets(desired_state_without_secrets)
+    logging.debug(f"Applying desire state: {desired_state_without_secrets}")
```
This `desired_state_without_secrets` is not being used.

cathay4t replied:
It is used by logging.

ffmancera replied:
Oh you are right. My bad.

ffmancera commented on libnmstate/netinfo.py:
```diff
@@ -35,10 +35,14 @@ def show(*, include_status_data=False):
+            show_with_plugins(
+                plugins,
+                include_status_data=include_status_data,
+                include_secret=include_secret,
```
What about `include_secrets`?

ffmancera commented on libnmstate/netinfo.py:
```diff
@@ -35,10 +35,14 @@ def show(*, include_status_data=False):
 
 
-def show_running_config():
+def show_running_config(include_secret=False):
```
What about `include_secrets`?

ffmancera reviewed (changes requested):
What would happen when Nmstate fails in a verification of an state that doesn't include 802.1X secrets? Will the current state report secrets anyway?

cathay4t commented:
> What would happen when Nmstate fails in a verification of an state that doesn't include 802.1X secrets? Will the current state report secrets anyway?

No. The `state_for_verify()` will never include secrets.

ffmancera reviewed (changes requested):
This is not fixing verification, the state is being gathered on `_verify_change()` in `netapplier.py`. Then verified using the `net_state.verify(current_state)`. In this case, the current_state will never contain the secrets and they are not going to be verified correctly.

cathay4t committed:
802.1x: Hide the password from output by default

Introduced new argument `include_secret=False|True` to:
    * `libnmstate.show_running_config()`
    * `libnmstate.show()`

This will mask the passwords with `<_password_hid_by_nmstate>` by
default(`include_secret=False`).

The passwords are also masked in logging and error message.

Added new argument:
    * `nmstatectl show -s `

Without `-s` switch, nmstatectl output will use the masked password.

When applying with above reserved password, nmstate will ignore it and
use current password.

Integration and unit test cases included.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera commented on tests/integration/nm/ieee802_1x_test.py:
```diff
@@ -134,11 +135,82 @@ def test_eth_with_802_1x(ieee_802_1x_env):
+    raises=NmstateLibnmError,
+    strict=False,
+)
+def test_apply_ieee_802_1x_with_reserved_password(ieee_1x_cli_up):
+    desire_state = show_only((TEST_1X_CLI_NIC,), include_secrets=False)
+
+    libnmstate.apply(desire_state)
+
+    current_iface_state = show_only((TEST_1X_CLI_NIC,), include_secrets=True)[
+        Interface.KEY
+    ][0]
+
+    assert (
+        TEST_PRIVATE_KEY_PASSWORD
+        == current_iface_state[Ieee8021X.CONFIG_SUBTREE][
+            Ieee8021X.PRIVATE_KEY_PASSWORD
+        ]
```
Please, could you create a test checking the current state does not contain the secrets if `include_secrets` is `False`?

Thanks!

ffmancera reviewed (changes requested):
Just one last test requirement. Thanks for all the work!

ffmancera reviewed (approved):
LGTM, thanks for all the work!
